src/annotator/integrations/index.js

```diff
@@ -9,26 +11,37 @@ import {
 /**
  * @typedef {import('../../types/annotator').Annotator} Annotator
  * @typedef {import('../../types/annotator').Integration} Integration
+ * @typedef {'main'|`sub_${string}`} FrameIdentifier - role based frame identifier
  */
 
 /**
  * Create the integration that handles document-type specific aspects of
- * guest functionality.
+ * guest functionality. Also returns a frame identifier based on the
+ * role of the frame: 'main' for the frame that holds the primary annotatable
+ * content, or 'sub_[random string]' for other annotatable frames.
  *
  * @param {Annotator} annotator
- * @return {Integration}
+ * @param {string|null} [frameIdentifier]
+ * @return {[Integration, FrameIdentifier]}.
```

Comment:
Setting (internally) a different frame identifier value from the value of `subFrameIdentifier` in the configuration is a good idea to avoid conflicts.

Currently, the `subFrameIdentifier` configuration property is only used to decide if the sidebar and notebook should be shown or not in that frame. It is kind of a bad naming. It would have been much better to rename it `enableSidebar` and transform it into a boolean. This is, nonetheless, out of question now.

External users have made use of `subFrameIdentifier` configuration property to force the sidebar from showing. Here is an example of that use in the [Page containing a cross-origin guest iframe development scenario](http://localhost:3000/document/cross-origin-iframe): 

https://github.com/hypothesis/client/blob/4d145d7d52e3d3b2b7041dbd4e53cdb7af7273c9/dev-server/documents/html/doyle-embedded.mustache#L8-L11

If a user decide to set the value of the `subFrameIdentifier` to 'main', it will result in errors, possible duplication of page notes or annotations.

Reply:
I think we could get away with deprecating `subFrameIdentifier` and treating existing uses of it as a boolean - if present, run client in "guest only" mode, otherwise run in host + guest mode. We can map deprecated properties to modern equivalents as part of configuration parsing.

Reply:
>I think we could get away with deprecating subFrameIdentifier and treating existing uses of it as a boolean - if present, run client in "guest only" mode, otherwise run in host + guest mode. We can map deprecated properties to modern equivalents as part of configuration parsing.

I think it is a good idea.